src/js/components/TaskFileDownloadComponent.jsx

```diff
@@ -0,0 +1,98 @@
+var React = require("react/addons");
+var classNames = require("classnames");
+
+var MesosActions = require("../actions/MesosActions");
+var MesosEvents = require("../events/MesosEvents");
+var MesosStore = require("../stores/MesosStore");
+
+function matchFileName(name) {
+  return (file) => file.name === name;
+}
+
+var TaskFileLinkComponent = React.createClass({
+  displayName: "TaskFileLinkComponent",
+  propTypes: {
+    children: React.PropTypes.node,
+    className: React.PropTypes.string,
+    fileName: React.PropTypes.string.isRequired,
+    task: React.PropTypes.object.isRequired
+  },
+
+  getInitialState: function () {
+    return {
+      file: this.getFile(),
+      requested: false
```

Comment:
Also, I personally really like using `is` / `has` prefixes for booleans where appropriate.